src/main/java/de/tum/in/www1/artemis/service/scheduled/AthenaScheduleService.java

```diff
@@ -0,0 +1,112 @@
+package de.tum.in.www1.artemis.service.scheduled;
+
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.ScheduledFuture;
+
+import javax.annotation.PostConstruct;
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.enumeration.ExerciseLifecycle;
+import de.tum.in.www1.artemis.repository.TextExerciseRepository;
+import de.tum.in.www1.artemis.security.SecurityUtils;
+import de.tum.in.www1.artemis.service.ExerciseLifecycleService;
+import de.tum.in.www1.artemis.service.ProfileService;
+import de.tum.in.www1.artemis.service.connectors.athena.AthenaSubmissionSendingService;
+
+@Service
+@Profile("athena & scheduling")
+public class AthenaScheduleService {
+
+    private final Logger log = LoggerFactory.getLogger(AthenaScheduleService.class);
+
+    private final ExerciseLifecycleService exerciseLifecycleService;
+
+    private final TextExerciseRepository textExerciseRepository;
+
+    private final ProfileService profileService;
+
+    private final Map<Long, ScheduledFuture<?>> scheduledAthenaTasks = new HashMap<>();
+
+    private final AthenaSubmissionSendingService athenaSubmissionSendingService;
+
+    public AthenaScheduleService(ExerciseLifecycleService exerciseLifecycleService, TextExerciseRepository textExerciseRepository, ProfileService profileService,
+            AthenaSubmissionSendingService athenaSubmissionSendingService) {
+        this.exerciseLifecycleService = exerciseLifecycleService;
+        this.textExerciseRepository = textExerciseRepository;
+        this.profileService = profileService;
+        this.athenaSubmissionSendingService = athenaSubmissionSendingService;
+    }
+
+    /**
+     * Schedule Athena tasks for all text exercises with future due dates on startup.
+     */
+    @PostConstruct
+    public void scheduleRunningExercisesOnStartup() {
+        if (profileService.isDev()) {
+            // only execute this on production server, i.e. when the prod profile is active
+            // NOTE: if you want to test this locally, please comment it out, but do not commit the changes
+            return;
+        }
+        final List<TextExercise> runningTextExercises = textExerciseRepository.findAllAutomaticAssessmentTextExercisesWithFutureDueDate();
+        runningTextExercises.forEach(this::scheduleExerciseForAthena);
```

Comment:
I know that we will only get SEMI-AUTOMATIC exercises from the database, however since these are two different values I'd still check is suggestions are actually enabled.
```suggestion
        runningTextExercises.forEach(this::scheduleExerciseForAthenaIfRequired);
```